base/src/main/java/io/spine/util/Preconditions2.java

```diff
@@ -37,31 +44,231 @@ private Preconditions2() {
     /**
      * Ensures that the passed string is not {@code null}, empty or blank string.
      *
-     * @param stringToCheck the string to check
+     * @param str
+     *         the string to check
      * @return the passed string
-     * @throws NullPointerException if the passed string is {@code null}
-     * @throws IllegalArgumentException if the string is empty or blank
+     * @throws IllegalArgumentException
+     *         if the string is empty or blank
+     * @throws NullPointerException
+     *         if the passed string is {@code null}
      */
     @CanIgnoreReturnValue
-    public static String checkNotEmptyOrBlank(String stringToCheck) {
-        return checkNotEmptyOrBlank(stringToCheck, "");
+    public static String checkNotEmptyOrBlank(String str) {
+        return checkNotEmptyOrBlank(str, "Encountered: \"%s\".", str);
     }
 
     /**
      * Ensures that the passed string is not {@code null}, empty or blank string.
      *
-     * @param stringToCheck the string to check
+     * @param str
+     *         the string to check
+     * @param errorMessageTemplate
+     *         the exception message template to use if the check fails
+     * @param errorMessageArgs
+     *         the arguments to be substituted into the message template
      * @return the passed string
-     * @throws NullPointerException if the passed string is {@code null}
-     * @throws IllegalArgumentException if the string is empty or blank
+     * @throws IllegalArgumentException
+     *         if the string is empty or blank
+     * @throws NullPointerException
+     *         if the passed string is {@code null}
      */
     @CanIgnoreReturnValue
-    public static String checkNotEmptyOrBlank(String stringToCheck, String message) {
-        checkNotNull(stringToCheck);
-        checkNotNull(message);
-        checkArgument(!stringToCheck.isEmpty(), message);
-        String trimmed = stringToCheck.trim();
-        checkArgument(trimmed.length() > 0, message);
-        return stringToCheck;
+    public static String checkNotEmptyOrBlank(String str,
+                                              @Nullable String errorMessageTemplate,
+                                              @Nullable Object @Nullable ... errorMessageArgs) {
+        checkNotNull(str, errorMessageTemplate);
```

Comment:
Shouldn't we pass `errorMessageArgs` as well?

Reply:
Please see in the latest code.

Reply:
I see that in our shortcut method we are passing the template string with parameter:
```
checkNotEmptyOrBlank(str, "Non-empty and non-blank string expected. Encountered: \"%s\".", str);
```
In this case, we'll have the `notNull` message equal to `"Non-empty and non-blank string expected. Encountered: \"%s\"."`, without the parameter being substituted with `null`, because we do not pass the parameters. Am I right?